includes/class-indieauth-token-endpoint.php

```diff
@@ -192,56 +228,89 @@ public function post( $request ) {
 		return new WP_OAuth_Response( 'invalid_request', __( 'Invalid Request', 'indieauth' ), 400 );
 	}
 
+
+	// Refresh Token Grant Type.
+	public function refresh_token( $params ) {
+
+		$diff = array_diff( array( 'refresh_token' ), array_keys( $params ) );
+		if ( ! empty( $diff ) ) {
+			return new WP_OAuth_Response( 'invalid_request', __( 'The request is missing one or more required parameters', 'indieauth' ), 400 );
+		}
+		$refresh = $this->refresh_tokens->get( $params['refresh_token'] );
+		if ( ! $refresh ) {
+			return new WP_OAuth_Response( 'invalid_grant', __( 'Invalid Token', 'indieauth' ), 400 );
+		}
+
+		// Destroy the refresh token.
+		$this->refresh_tokens->destroy( $params['refresh_token'] );
```

Comment:
This may be happening too early, i.e. if the below fails?

Reply:
If it fails, shouldn't it require the full flow?